lib/trino-parquet/src/main/java/io/trino/parquet/writer/valuewriter/UuidValueWriter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.writer.valuewriter;
+
+import io.airlift.slice.Slice;
+import io.trino.spi.block.Block;
+import io.trino.spi.type.Type;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+
+import static java.util.Objects.requireNonNull;
+
+public class UuidValueWriter
```

Comment:
The difference is in `slice.toByteBuffer()` vs `slice.getBytes()`.

`toByteBuffer` doesn't require copying the data and is supported when slice is byte[]-backed
`getBytes` does a copy, but works for any Slice. 

in case of UUID, slices are long[]-backed.